integration/test/test_hint_time.py

```diff
@@ -0,0 +1,182 @@
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+"""HINT_TIME
+
+"""
+
+import sys
+import unittest
+import os
+import glob
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from integration.test import geopm_context
+import geopmpy.io
+import geopmpy.error
+
+from integration.test import util
+from integration.experiment import machine
+if util.do_launch():
+    # Note: this import may be moved outside of do_launch if needed to run
+    # commands on compute nodes such as geopm_test_launcher.geopmread
+    from integration.test import geopm_test_launcher
+    geopmpy.error.exc_clear()
+
+class AppConf(object):
+    """Class that is used by the test launcher in place of a
+    geopmpy.io.BenchConf when running the hint_time benchmark.
+
+    """
+    def write(self):
+        """Called by the test launcher prior to executing the test application
+        to write any files required by the application.
+
+        """
+        pass
+
+    def get_exec_path(self):
+        """Path to benchmark filled in by template automatically.
+
+        """
+        bin_name = 'test_hint_time'
+        # Look for in place build
+        script_dir = os.path.dirname(os.path.realpath(__file__))
+        bin_path = os.path.join(script_dir, '.libs', bin_name)
+        if not os.path.exists(bin_path):
+            # Look for out of place build from using apps/build_func.sh
+            int_dir = os.path.dirname(script_dir)
+            bin_path_op = os.path.join(int_dir, 'build/integration/test/.libs', bin_name)
+            if not os.path.exists(bin_path_op):
+                msg = 'Could not find application binary, tried \n    "{}"\n    "{}"'.format(
+                      bin_path, bin_path_op)
+                raise RuntimeError(msg)
+            bin_path = bin_path_op
+        return bin_path
+
+    def get_exec_args(self):
+        """Returns a list of strings representing the command line arguments
+        to pass to the test-application for the next run.  This is
+        especially useful for tests that execute the test-application
+        multiple times.
+
+        """
+        return []
+
+
+class TestIntegration_hint_time(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Create launcher, execute benchmark and set up class variables.
+
+        """
+        sys.stdout.write('(' + os.path.basename(__file__).split('.')[0] +
+                         '.' + cls.__name__ + ') ...')
+        cls._test_name = 'hint_time'
+        cls._report_path = 'test_{}.report'.format(cls._test_name)
+        cls._trace_path = 'test_{}.trace'.format(cls._test_name)
+        cls._image_path = 'test_{}.png'.format(cls._test_name)
+        cls._skip_launch = not util.do_launch()
+        cls._agent_conf_path = 'test_' + cls._test_name + '-agent-config.json'
+        # Clear out exception record for python 2 support
+        geopmpy.error.exc_clear()
```

Comment:
I think we can remove all of this now, including from the template.

Reply:
There's a few other bits re: Python 2.  I suppose we should just save all of this for a future clean-up task.